client/src/pages/AddTour.js

```diff
@@ -32,6 +33,32 @@ export default function AddTour() {
 
   const concurrentTour = query.get("type") === "concurrent";
 
+  useEffect(() => {
+    if (id) {
+      const doFetch = async () => {
+        try {
+          const data = await getDataByID({
+            collectionName: "tasks",
+            id: id,
+          });
+          setName(data.name);
+          setStart(data.start);
+          setDest(data.dest);
+          setDate(data.date);
+          setAssignment(data.assignment);
+          setCargo(data.cargo);
+          setPriority(data.priority);
+          setInfo(data.info);
+          setCheckboxes(data.checkboxes);
+          setWeekDays(data.weekDays);
+        } catch (e) {
+          console.error(e);
```

Comment:
Consider writing the error in a state and render it in the UI. Otherwise, users won't notice :).

Reply:
Changes will be dealt with in #116.